### PR TITLE
Bump go1.21 and update dependencies

### PR DESCRIPTION
**Pull request by jkhelil**
# Changes
- Bump go to 1.21
- Bump sigs.k8s.io/controller-runtime to 0.14.6
- Bump k8s.io/api to 0.27.4
- Update OWNERS

# Submitter Checklist

- [ ] Includes tests if functionality changed/was added
- [ ] Includes docs if changes are user-facing
- [ ] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)
- [ ] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/main/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes
NONE

### CODE REVIEW

**Comment by jkhelil** [comment]
@shipwright-io/operator-approvers PTAL

File: go.mod [hunk 1]
```diff
@@ -1,103 +1,114 @@
...
-	k8s.io/api v0.23.5
-	k8s.io/apiextensions-apiserver v0.23.4
-	k8s.io/apimachinery v0.23.5
-	k8s.io/client-go v0.23.5
-	sigs.k8s.io/controller-runtime v0.11.0
+	github.com/tektoncd/operator v0.68.0
+	k8s.io/api v0.27.4
+	k8s.io/apiextensions-apiserver v0.27.4
+	k8s.io/apimachinery v0.27.4
+	k8s.io/client-go v1.5.2
```
Comment (adambkaplan) [inline go.mod:15]: Please ensure client-go is aligned with the other k8s versions. This will break things otherwise.

Comment (jkhelil) [reply go.mod:15]: Yes there is a replcaement directive for client-go 
k8s.io/client-go => k8s.io/client-go v0.26.8
the k8s.io/client-go v1.5.2 is forced by go mod algorithm
I can see the same thing for the tekton operator go mod https://github.com/tektoncd/operator/blob/main/go.mod#L33

Also, it is worth adding that i tested it locally and it is working correctly

**Comment by jkhelil** [comment]
@shipwright-io/operator-reviewers 
/assign @adambkaplan

**Pull request closed without merging**